### PR TITLE
Don't create kprobes with more than 128 args in auditbeat's guess subsystem

### PR DESCRIPTION
**Pull request by fearful-symmetry**
## Proposed commit message
This is a little silly. We have a few kprobes in auditbeat that will attempt to create up to _256_ args to read the return value from the register of a kretprobe; according to the kernel docs, only 128 args are supported. On older kernel releases, it would silently cut off any args greater than 128 (meaning this code never really worked as intended), but sometime around kernel 6.9, the kernel will throw an oops: https://bugzilla.redhat.com/show_bug.cgi?id=2303876. On older kernels, this technically doesn't change anything, as the kernel would just throw out any arguments over 128.

This modifies the kprobe creation so we're not trying to read from over 128 args.

If you want to test this behavior, you'll need Fedora, Arch, or another edge distro. The link above has some C code to reproduce the issue.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

File: x-pack/auditbeat/module/system/socket/guess/inetsock6.go [hunk 1]
```diff
@@ -191,7 +192,7 @@ func (g *guessInetSockIPv6) Probes() (probes []helper.ProbeDef, err error) {
 	raddrOffsets := g.offsets
 	g.offsets = nil
 	const sizePtr = int(sizeOfPtr)
-	var fetch []string
+	fetch := make([]string, 0)
```
Comment (nicholasberlin) [inline x-pack/auditbeat/module/system/socket/guess/inetsock6.go:195]: Why this change? (and the others like it)

Comment (fearful-symmetry) [reply x-pack/auditbeat/module/system/socket/guess/inetsock6.go:195]: It's entirely the linter; it complains if you do something like `var list []string`, since it wants you to pre-allocate the array based on size, but for most of these, you don't actually know the size of the array in advance, so `make()` just pre-allocates it to zero.

**Review by nicholasberlin — APPROVED** [review]

File: x-pack/auditbeat/module/system/socket/guess/inetsock6.go (hunk 1 above)
Comment (efd6) [reply x-pack/auditbeat/module/system/socket/guess/inetsock6.go:195]: This is one of the multitude of cases where the linter pushes people into writing worse code. IMO we should push back against this. To demonstrate why, this is the comparison between the generated code from the two approaches: https://godbolt.org/z/1zs7fY5f7.

Comment (nicholasberlin) [reply x-pack/auditbeat/module/system/socket/guess/inetsock6.go:195]: I clicked that link, and the `make` version was being compiled as `x86 gc 1.20` versus `x86-64 gc 1.22.1`. After switching the `make` version to be the same, I see this diff:
Screenshot 2024-08-14 at 7 24 22 AM

They seem pretty close to me. Am I missing something? (I probably am)

Comment (nfritts) [reply x-pack/auditbeat/module/system/socket/guess/inetsock6.go:195]: This is pretty interesting.  They changed something in 1.21 (released Aug 2023).  If you compare 1.20 or earlier you see that the lint preferred version generates much more code.  Starting with 1.21 they appear to be equivalent.

Comment (efd6) [reply x-pack/auditbeat/module/system/socket/guess/inetsock6.go:195]: Nice that the compiler is able to optimise this to the same bytes. @nicholasberlin, no you weren't missing anything, @nfritts has summarised perfectly. I guess despite [this](https://go.dev/wiki/CodeReviewComments#declaring-empty-slices) the gc team has made an effort to recognise and re-write the non-idiomatic phrasing.
